In [48]:
import pandas as pd
from pandas import read_csv
from pandas import Series



In [52]:
data1= pd.read_csv("C:/Users/cgdeer/Downloads/HERDDATA/HERD16wid.csv", index_col='UNITID')


In [53]:
len(data1)


642

In [54]:
data2= read_csv("C:/Users/cgdeer/Downloads/HERDDATA/GeoC1617.csv", index_col='UNITID')


In [55]:
len(data2)

7521

In [58]:
Merged = pd.merge(data1, data2, how='inner', on="UNITID")


In [60]:
len(Merged)


621

appears to remove entries that do not have a UNITID, such as Ranks 48,65,71,72,148,218,242,255,304,368,496,552.  Also removes 7 end lines, total removed was 19. 621+19=640.  One Line removed from top.  641.  Suspect one dupe.

In [59]:
Merged.to_csv("C:/Users/cgdeer/Downloads/HERDDATA/Merged.csv")